ntfysh_client/NotificationDialog.cs

```diff
@@ -0,0 +1,263 @@
+                return;
+            }
+            this.progress = (int)((this._timeout - this.shownStopwatch.ElapsedMilliseconds) * 100 / this._timeout);
+            this.lbTimeout.Text = $"{(int)(this._timeout - this.shownStopwatch.ElapsedMilliseconds) / 1000}";
+        }
+
+        protected override void SetVisibleCore(bool value)
+        {
+            this.SetWindowPosition();
+            if (value)
+            {
+                this.BringToFront();
+                AnimateWindow(
+                    this.Handle,
+                    time: 250,
+                    flags: NFWinUserAnimateWindowConstnats.AW_SLIDE | NFWinUserAnimateWindowConstnats.AW_VER_NEGATIVE
+                );
+            }
+            base.SetVisibleCore(value);
+        }
+
+        protected override void OnShown(EventArgs e)
+        {
+            base.OnShown(e);
+        }
+
+        private void SetWindowPosition()
+        {
+            int workingtop = Screen.PrimaryScreen.WorkingArea.Height - this.Height;
+            this.Top = workingtop - NotificationDialog.ScreenMargin;
+
+            int workingleft = Screen.PrimaryScreen.WorkingArea.Width - this.Width;
+            this.Left = workingleft - NotificationDialog.ScreenMargin;
+        }
+
+        private void ui_hide_window(object? sender, EventArgs? e)
+        {
+            AnimateWindow(
+                this.Handle,
+                time: 250,
+                flags: NFWinUserAnimateWindowConstnats.AW_SLIDE | NFWinUserAnimateWindowConstnats.AW_VER_POSITIVE | NFWinUserAnimateWindowConstnats.AW_HIDE
+            );
+            this.IsVisible = false;
+        }
+
+        private void handleTimeout(object? sender, EventArgs? e)
+        {
+            this.cancelTimer();
+            if (this.InvokeRequired)
+            {
+                // on a background thread, so invoke on the UI thread
+                this.Invoke(new Action(() => this.ui_hide_window(sender, e)));
+            }
+            else
+            {
+                // in the UI thread, invoke directly
+                this.ui_hide_window(sender, e);
+            }
+        }
+
+        private Image? ConvertToolTipIconToImage(ToolTipIcon icon)
+        {
+            switch (icon)
+            {
+                case ToolTipIcon.Info:
+                    return SystemIcons.Information.ToBitmap();
+                case ToolTipIcon.Warning:
+                    return SystemIcons.Warning.ToBitmap();
+                case ToolTipIcon.Error:
+                    return SystemIcons.Error.ToBitmap();
+                case ToolTipIcon.None:
+                default:
+                    return null;
+            }
+        }
+
+        private void InitializeWindowHidden()
+        {
+            this.Opacity = 0;
+            this.ShowNotification("Title", "Message");
+            this.IsVisible = false;
+            this.Opacity = 1;
+        }
+
+        private void btnClose_Click(object sender, EventArgs e)
```

Comment:
I try to rename my elements to match the TitleCase naming convention. Not a major one, just a nice-to-have. I appreciate this isn't consistent within the application anyway.

Reply:
settings dialog is using camelCase for everything....

I think the Pep8 guidance that "consistency within a file is most important" is a good point here.

I opted to make the new things in SettingsDialogue always use camelCase, but changed the NotificationDialogue be TitleCase.